starport/services/serve/plugin.go

```diff
@@ -0,0 +1,92 @@
+package starportserve
+
+import (
+	"context"
+	"strings"
+
+	"github.com/tendermint/starport/starport/pkg/cmdrunner/step"
+	"github.com/tendermint/starport/starport/pkg/gomodule"
+	starportconf "github.com/tendermint/starport/starport/services/serve/conf"
+)
+
+// TODO omit -cli log messages for Stargate.
+
+type Plugin interface {
```

Comment:
We may need to re-think about Plugin's methods in future to make its APIs more understandable and find a nice balance on the shared code between _starportserve.Serve()_ and a Plugin's implementation.